### ISSUE TITLE
can not produce hdf5 file when used new data to test the model

### ISSUE THREAD

**Issue by sunchuance** [DeepRank/deeprank#250]
**Describe the bug**
At present I have trained a model, but when I test it with new data, there is a problem. The problem is that the hdf5 file generated by the new data cannot be processed in the next step (The script to generate the hdf5 file comes from deeprank/paper_pretrained_models/docking_models_scoring/test.py in github).The specific information is in the pdf file. Can you tell me why? Thank you very much.
**Environment:**
- OS system:
- Version:
- Branch commit ID: 
- Inputs:

**To Reproduce**
Steps/commands to reproduce the behaviour:
 1.  
 2.
 3. 

**Expected Results**

**Actual Results or Error Info**

**Additional Context**
[Untitled1 - Jupyter Notebook.pdf](https://github.com/DeepRank/deeprank/files/8096960/Untitled1.-.Jupyter.Notebook.pdf)

**Comment by sunchuance** [comment]
Thank you very much for your reply. Which of the two scoring values given by the model should be the main one?as the picture shows.
![图片1](https://user-images.githubusercontent.com/77307893/155042171-bff3c650-2111-4c0b-8e9e-59e401ef75bd.png)

**Comment by CunliangGeng** [comment]
Hi @sunchuance, it seems you're using an old version of DeepRank. Please use the the latest one (https://pypi.org/project/deeprank/0.2.1/), this would solve the error you met.

**Comment by CunliangGeng** [comment]
> Thank you very much for your reply. Which of the two scoring values given by the model should be the main one?as the picture shows. ![图片1](https://user-images.githubusercontent.com/77307893/155042171-bff3c650-2111-4c0b-8e9e-59e401ef75bd.png)

I didn't get your point. How did you get these values? Could you paste the commands?

**Comment by sunchuance** [comment]
> > Thank you very much for your reply. Which of the two scoring values given by the model should be the main one?as the picture shows. ![图片1](https://user-images.githubusercontent.com/77307893/155042171-bff3c650-2111-4c0b-8e9e-59e401ef75bd.png)
> 
> I didn't get your point. How did you get these values? Could you paste the commands?

Thanks for your reply, I understand the prediction score now, but I now encounter a situation where there are two ways to divide the training set and the test set, as shown in the figure below. When I use proportional division, the loss function can be decreased, but the result is very poor when the obtained model predicts a batch of new data. When I use the directly defined training set and test set, the loss function of the test set does not decrease. I would like to ask what is the difference between these two ways
![图片1](https://user-images.githubusercontent.com/77307893/155700521-de4c8e67-a2e1-4ffe-846a-66d417492b79.png)
![图片2](https://user-images.githubusercontent.com/77307893/155700528-70a3e2a2-9f8b-40da-bf29-2c5062e9103c.png)

**Comment by CunliangGeng** [comment]
The only difference is if you prefer manually or automatically preparing the training, validation and test datasets. For automatic way, use `divide_trainset` parameter; otherwise, use `train_database, valid_database, test_database` parameters.

**Comment by sunchuance** [comment]
> The only difference is if you prefer manually or automatically preparing the training, validation and test datasets. For automatic way, use `divide_trainset` parameter; otherwise, use `train_database, valid_database, test_database` parameters.

Ok, thank you, in addition, the last question is about the output of two columns for scoring. According to the article, one column is close to native and the other column is close to wrong. Does that mean that the one on the left is close to wrong?
![图片3](https://user-images.githubusercontent.com/77307893/155708538-8ff63862-a21b-46ab-95e8-da34d48226d3.png)

**Comment by CunliangGeng** [comment]
> > The only difference is if you prefer manually or automatically preparing the training, validation and test datasets. For automatic way, use `divide_trainset` parameter; otherwise, use `train_database, valid_database, test_database` parameters.
> 
> Ok, thank you, in addition, the last question is about the output of two columns for scoring. According to the article, one column is close to native and the other column is close to wrong. Does that mean that the one on the left is close to wrong? ![图片3](https://user-images.githubusercontent.com/77307893/155708538-8ff63862-a21b-46ab-95e8-da34d48226d3.png)

how did you get the scoring values? can you show the commands?

**Comment by sunchuance** [comment]
> > > The only difference is if you prefer manually or automatically preparing the training, validation and test datasets. For automatic way, use `divide_trainset` parameter; otherwise, use `train_database, valid_database, test_database` parameters.
> > 
> > 
> > Ok, thank you, in addition, the last question is about the output of two columns for scoring. According to the article, one column is close to native and the other column is close to wrong. Does that mean that the one on the left is close to wrong? ![图片3](https://user-images.githubusercontent.com/77307893/155708538-8ff63862-a21b-46ab-95e8-da34d48226d3.png)
> 
> how did you get the scoring values? can you show the commands?

OK. This is commands.After this command, a file (test_output.hdf5) is obtained, and the output in it is the score.
![图片5](https://user-images.githubusercontent.com/77307893/155715070-e5512709-e41d-4857-a563-d9f9020628dd.png)

![图片4](https://user-images.githubusercontent.com/77307893/155714657-bb4ce42b-32cb-4678-890e-a4804e332be4.png)

**Comment by CunliangGeng** [comment]
Thanks! By default, we use 1 to label the near-native class and 0 to label the wrong class. So you're right that the 1st column is for class 0.

**Comment by sunchuance** [comment]
> Thanks! By default, we use 1 to label the near-native class and 0 to label the wrong class. So you're right that the 1st column is for class 0.
Ok, thanks, and I would like to ask if the pre-trained model in the paper can score the different decoy of the antibody.

**Comment by CunliangGeng** [comment]
Hi @sunchuance  our pre-trained model was not trained using antibody structures, so it's not advised to score antibodies.
You could find more research details about DeepRank in our publication https://www.nature.com/articles/s41467-021-27396-0
